### PR TITLE
Wind pushes smoke/gas/fire/hot air.

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Wind will now affect the spread of smoke/gas/fire/hot air"```

#### Purpose of change
With the increasing wind changes, it seems logical that gases and fire would be influenced by it.

#### Describe the solution
If the fire or gas or smoke is not indoors, not sheltered from the wind, and the wind is over 5mph, then it's spread direction will be influenced by the wind direction. 
High winds will make fires burn hotter and shorter and spread more.
More smoke will be produced and the smoke will blow in the same direction as the wind.
This applies to poison gas, tear gas, cigarette smoke, nuclear bomb clouds, hot air, and so on.
CPU concerns limit long smoke columns and trails, and the reality bubble would limit that anyway.
If you stand upwind, and set a fire, you may well be safe from it's spread. 
Once again these changes may be best served by the wind direction and power being displayed on the sidebar, which @nsklaus is working on.

#### Describe alternatives you've considered
N/A
#### Additional context
torching the starting shelter surrounded by fungal growth, with a fairly strong easterly wind. the fire is spreading mostly west and north and south. the smoke too.

image

The fire code mangled my brain a bit with it's complexity and endless indents and lambdas, apologies if I've misunderstood any of it.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
I'm not sure why this makes me so happy, but this makes me *so happy*.

File: src/field.cpp [hunk 1]
```diff
@@ -1026,7 +1108,66 @@ bool map::process_fields_in_submap( submap *const current_submap,
                         // Below we will access our nearest 8 neighbors, so let's cache them now
                         // This should probably be done more globally, because large fires will re-do it a lot
                         auto neighs = get_neighbors( p );
-
+                        // get the neighbours that are allowed due to wind direction
+                        double raddir = ( ( winddirection + 180 ) % 360 ) * ( M_PI / 180 );
```
Comment (Brambor) [inline src/field.cpp]: Isn't that the same code as at lines 686 (at least the following 20 lines)?
Couldn't it be made into a function?

File: src/field.cpp [hunk 2]
```diff
@@ -1026,7 +1108,66 @@ bool map::process_fields_in_submap( submap *const current_submap,
...
+                            roundedy = 1;
+                        } else if( fy < -0.5 ) {
+                            roundedy = -1;
+                        } else {
+                            roundedy = 0;
+                        }
+                        tripoint removepoint( p - point( roundedx, roundedy ) );
+                        tripoint removepoint2;
+                        tripoint removepoint3;
+                        if( roundedy != 0 && roundedx == 0 ) {
```
Comment (Brambor) [inline src/field.cpp]: Again, feels like the same as lines 719 through 737.

**Comment by davidpwbrown** [comment]
Marking as WIP, as there is a new wind system using debug overrides and fixing various things, that will make this obsolete as it is, itll need to be rebased to #28143

**Comment by davidpwbrown** [comment]
rebased to master, using the new wind override system

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/everburning-fires/19415/9

**Pull request merged**